tapscript/util.go

```diff
@@ -36,6 +36,18 @@ func PayToTaprootScript(taprootKey *btcec.PublicKey) ([]byte, error) {
 		Script()
 }
 
+// FlipParity turns the given public key from even to odd parity or vice versa.
+func FlipParity(evenParityKey *btcec.PublicKey) *btcec.PublicKey {
+	keyCompressed := evenParityKey.SerializeCompressed()
+	keyCompressed[0] ^= 1
+
+	// We already know the even key is a valid point on the curve, so we
+	// don't need to check the error here as the flipped key will also be
+	// valid.
+	oddKey, _ := btcec.ParsePubKey(keyCompressed)
+	return oddKey
```

Comment:
You use the names `evenParityKey` and `oddKey` here. But they could be generalised further because the argument key doesn't have to be even. `evenParityKey -> key`, and `oddKey -> flippedParityKey` ?

The comment in this function says "We already know the even key is a valid point on the curve". If this function is to be used generally, we might not know that. So we might want the comment to state an assumption here. Or check the error.

Reply:
Oops, fixed.